### PR TITLE
Change the mfem dependency to a recent git SHA

### PR DESCRIPTION
**Pull request by jandrej**
This is here for discussion. I tested the setup on LC.

### CODE REVIEW

**Review by white238 — APPROVED** [review]

File: scripts/uberenv/packages/serac/package.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ class Serac(CMakePackage):
     depends_on('py-sphinx', when="+devtools")
 
     # Libraries that support +debug
-    debug_deps = ["mfem@4.1.0~shared+hypre+metis+superlu-dist+lapack+mpi",
+    debug_deps = ["mfem@4.1.0p1~shared+hypre+metis+superlu-dist+lapack+mpi",
```
Comment (jamiebramwell) [inline scripts/uberenv/packages/serac/package.py:99]: Things like this are the stuff of nightmares. Thank you for figuring it out!

**Review by jamiebramwell — APPROVED** [review]

File: scripts/uberenv/packages/serac/package.py (hunk 1 above)
Comment (jandrej) [reply scripts/uberenv/packages/serac/package.py:99]: Credit is on @white238. He knew the right person to ask :)

**Comment by jamiebramwell** [comment]
LGTM

**Comment by jamiebramwell** [comment]
LGTM

**Pull request merged**